hyperspy/misc/utils.py

```diff
@@ -1639,3 +1639,43 @@ def display(obj):
         display.display(obj)
     except ImportError:
         print(obj)
+
+class TupleSA(tuple):
+    """A tuple that can set the attributes of its items
+    """
+    def __getitem__(self, *args, **kwargs):
+        item = super().__getitem__(*args, **kwargs)
+        try:
+            return type(self)(item)
+        except TypeError:
+            return item
```

Comment:
Add a comment? Something along the line of:
```python
# for single item case, e.g. a[0]
# TupleSA requires an iterable item
```

Reply:
Done